server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -554,11 +559,34 @@ def filter(
         columns: list[str],
         step: "FilterStep",
     ) -> StepContext:
+
+        # To handle relative date formats
+        if isinstance(step.condition, (ConditionComboAnd, ConditionComboOr)):
+            cond = step.condition
+
+            conditions = cond.and_ if isinstance(cond, ConditionComboAnd) else cond.or_
+
+            for sub_cond in conditions or []:
+                if isinstance(sub_cond, DateBoundCondition):
+                    if isinstance(sub_cond.value, RelativeDate):
+                        value_str_time = (
+                            evaluate_relative_date(sub_cond.value)
+                            .replace(tzinfo=timezone.utc)
```

Comment:
```suggestion
                            .astimezone(timezone.utc)
```

`replace` would overwrite the TZ info, which is not what we want: We need to re-localize the datetime object to UTC

Reply:
👍🏾 updated !